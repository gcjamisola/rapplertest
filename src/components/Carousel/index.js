import React, { Component } from 'react';
import PropTypes from 'prop-types';

import Slider from 'react-slick';

import {
  CarouselContainer,
  ReadMore,
  StyledSlider,
  SliderElement,
  SliderImage,
  SliderTitle,
  SliderDescription
} from '../../styles';

class Carousel extends Component {
  constructor(props) {
    super(props);

this.state = {
storyArray: []
}

    this._sliderElement = this._sliderElement.bind(this);
  }

  componentWillReceiveProps(nextProps){
    if(this.props.stories !== nextProps.stories) {
      this.setState({ storyArray: nextProps.stories});
    }
  }

  _sliderElement(arr) {
    return arr.map((value, key) => {
      const {
        full: imgUrl,
        title,
        metadesc: storyDesc } = value;

      return (
        <div key={key}>
          <SliderImage src={`${imgUrl}`} />
          <SliderElement>
            <SliderTitle>{`${title}`}</SliderTitle>
            <SliderDescription>{`${storyDesc}`}</SliderDescription>
            <ReadMore>> Read More...</ReadMore>
          </SliderElement>
        </div>
      );
    })
  }

  render() {
    const { settings } = this.props;
    const { storyArray } = this.state;

    return (
      <CarouselContainer>
        <StyledSlider>
            <Slider {...settings}>
              { this._sliderElement(storyArray)  }
            </Slider>
        </StyledSlider>
      </CarouselContainer>
    );
  }
}

Carousel.propTypes = {
  setting: PropTypes.object,
  stories: PropTypes.array
}

export default Carousel;
