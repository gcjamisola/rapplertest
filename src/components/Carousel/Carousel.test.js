import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Carousel from './index';

describe('<Carousel/>', () => {
    it('renders <Carousel/> component', () => {
        const component = shallow(<Carousel/>);
        const tree = toJson(component);
        expect(tree).toMatchSnapshot();
    });
});
