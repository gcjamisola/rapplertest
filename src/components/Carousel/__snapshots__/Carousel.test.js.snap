// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Carousel/> renders <Carousel/> component 1`] = `
<styled.div>
  <styled.div>
    <Slider />
  </styled.div>
</styled.div>
`;
